:py:mod:`pine.client.exceptions`
================================

.. py:module:: pine.client.exceptions

.. autoapi-nested-parse::

   PINE client exceptions module.



Module Contents
---------------

.. py:exception:: PineClientException(message: str, cause: Exception = None)

   Bases: :py:obj:`Exception`

   Base class for PINE client exceptions.
       

   Constructor.

   :param message: the message
   :type message: str
   :param cause: optional cause, defaults to ``None``
   :type cause: Exception, optional

   .. py:attribute:: message
      

      The message.

      :type: str



.. py:exception:: PineClientHttpException(method: str, path: str, resp: requests.Response)

   Bases: :py:obj:`PineClientException`

   A PINE client exception caused by an underlying HTTP exception.
       

   Constructor.

   :param method: the REST method (``"get"``, ``"post"``, etc.)
   :type method: str
   :param path: the human-readable path that caused the exception
   :type path: str
   :param resp: the :py:class:`Response <requests.Response>` with the error info
   :type resp: requests.Response

   .. py:attribute:: resp
      :annotation: :requests.Response

      The :py:class:`Response <requests.Response>` with the error info

      :type: requests.Response


   .. py:method:: status_code(self)
      :property:



.. py:exception:: PineClientValueException(obj: dict, obj_type: str)

   Bases: :py:obj:`PineClientException`

   A PINE client exception caused by passing invalid data.
       

   Constructor.

   :param obj: the error data
   :type obj: dict
   :param obj_type: human-readable type of object
   :type obj_type: str


.. py:exception:: PineClientAuthException(message: str, cause: Exception = None)

   Bases: :py:obj:`PineClientException`

   Base class for PINE client exceptions.
       

   Constructor.

   :param message: the message
   :type message: str
   :param cause: optional cause, defaults to ``None``
   :type cause: Exception, optional


